pydantic_ai_slim/pydantic_ai/_cli.py

```diff
@@ -182,11 +182,12 @@ def cli(args_list: Sequence[str] | None = None, *, prog_name: str = 'pai') -> in
             console.print('[red]Error: Agent must be specified in "module:variable" format[/red]')
             return 1
 
-    try:
-        agent.model = infer_model(args.model)
-    except UserError as e:
-        console.print(f'Error initializing [magenta]{args.model}[/magenta]:\n[red]{e}[/red]')
-        return 1
+    if agent.model is None:
```

Comment:
What if is not none and you pass a model via CLI? Shouldn't you be able to change the model the agent uses it without changing ur code?